docs/developer/release-cycle.md

```diff
@@ -0,0 +1,69 @@
+# Release Cycle
+
+The release cycle is the time between when we start working on a release on the `main` branch until the `.0` release (e.g. `v1.3.0`).
+
+**Note**: For the `v1.3` and `v1.4` release cycles we assume a ~ 4 months (~ 17 weeks) release cycle. This should be revisited after the `v1.4.0` release.
+
+The release cycle can be split up into the following phases:
+
+* Week 1-12: Active development
+    * Alpha releases based on the main branch can be released when necessary
+* Week 13-14:
+    * Beta releases based on the main branch
+    * The beta releases are meant as a preview what the upcoming release will look like
+    * Provider can start adopting the new release based on the beta releases
+    * Breaking changes to providers have to be discussed and approved in the office hours
+* Week 15-16:
+    * The release branch is created
+    * RC releases based on the release branch
+    * There is a feature freeze on the release branch
+    * Only bug fixes / documentation / tests fixes and additions are allowed
+* Week 17:
+    * `.0` GA release is created based on the release branch
+* After that:
+    * `.1+`: Monthly patch releases based on the release branch
+    * Provider releases corresponding to the CAPI minor release when they are ready
+
+Some additional notes:
+
+* Support for new Kubernetes releases is first implemented on the main branch, then cherry-picked into
+  supported release branches when feasible and eventually released in the next monthly patch release(s).
+* New Kubernetes and Controller-Runtime (CR) minor releases are only picked up on the main branch.
+
+The following diagram visualizes the release cycles with the v1.2, v1.3 and v1.4 releases:
+
+<!-- The release cycle png can be opened and edited in draw.io -->
+![Release Cycle Overview](release-cycle-overview.png)
+
+# v1.3 Release Cycle timeline
```

Comment:
Should we add a note that this release will be used to set up the work for the release team that will be effective from the following cycle?

Similarly, down I will add a note about the fact that 1.4 will be the first release cycle with a release team and also add a section with the release team Roles/Names (TBD)

Reply:
Mostly done.

I think it makes sense to have a table with Role => Names (including shadows) mapping. 
But I wouldn't block this PR until the release team is 100% finalized.

Reply:
sure, it should be a placeholder with TBD

Reply:
Will do on the next update